src/graphics.cpp

```diff
@@ -226,10 +301,39 @@ void Graphics::renderHUD(fCoord25D playerPos, std::array<int, 2> mapDimensions,
     for (int nx = 0; nx < minimapDimension; nx++) {
         for (int ny = 0; ny < minimapDimension; ny++) {
             int formula = (yStart+ny) * nMapHeight + (xStart+nx);
-            if ((formula) < 0 || (formula) > map.length() || (xStart+nx) < 0 || (xStart+nx) >= nMapHeight)
+            if ((formula) < 0 || (formula) > map.length() || (xStart+nx) < 0 || (xStart+nx) >= nMapHeight) {
                 mvaddch(ny + 1, nx, ' ');
-            else
-                mvaddch(ny + 1, nx, (chtype)map[formula]);
+            } else {
+                // This code sucks.
+                // TODO: Make it not suck as much.
+                char mapBlock = (chtype)map[formula];
+                int color = 1;
+                short character = 0x2588;
+                switch (mapBlock) {
+                case 'W':
+                    break;
+                case 'R':
+                    color = 2;
+                    break;
+                case 'G':
+                    color = 3;
+                    break;
+                case 'B':
+                    color = 4;
+                    break;
+                case 'Y':
+                    color = 5;
+                    break;
+                default:
+                    color = 1;
```

Comment:
Maybe these colors should have names in an enum like the brightness values?

Reply:
Oh they absolutely should.

Reply:
Done